pytorch_lightning/trainer/trainer.py

```diff
@@ -458,8 +458,14 @@ def train(self):
         self.train_loop.on_train_start()
 
         try:
+            start_epoch = self.current_epoch
+            end_epoch = self.max_epochs
+
             # run all epochs
-            for epoch in range(self.current_epoch, self.max_epochs):
+            for epoch in range(start_epoch, end_epoch):
+
+                # Update trainer current epoch
+                self.current_epoch = epoch
 
                 # reset train dataloader
```

Comment:
i'd prefer to push this logic to be in `on_train_epoch_start`. then we can leave the current epoch initialization as-is.

Reply:
Makes sense. I will makes that change